pkg/gui/main_panel.go

```diff
@@ -39,7 +51,25 @@ func (gui *Gui) scrollRightMain(g *gocui.Gui, v *gocui.View) error {
 	mainView := gui.getMainView()
 	ox, oy := mainView.Origin()
 
-	return mainView.SetOrigin(ox+gui.Config.UserConfig.Gui.ScrollHeight, oy)
+	content := mainView.BufferLines()
+	var largestNumberOfCharacters int
+	for _, txt := range content {
```

Comment:
I don't like this loop as it is unnecessary to go thru it every single time.
If there's some function that's already built-in that can do something similar only once, please point it out to me.
Thank you.

Reply:
Yeah I share your concern, this could be quite an expensive loop to run. What if we used `ViewBufferLines` instead so that we're only looping through the lines that are displayed? That way there will be far less to loop through

Reply:
Hmm...i am not sure how that would help. ViewBufferLines returns an []string too and i would most probably have to loop thru each to find the longest line. Similar to what i have already implemented.
I am going to need your help pointing out what i have misunderstood :)

Reply:
The difference is in the length of the array. Sometimes we are dealing with diffs that are thousands of lines long, in which case iterating through the entire array of BufferLines will be very expensive. On the other hand, iterating through just the ViewBufferLines is quite cheap because it's limited to the height of the window.

Reply:
Ok i understand. Changes coming soon